app/src/main/java/org/schabi/newpipe/util/ListUtils.java

```diff
@@ -0,0 +1,23 @@
+package org.schabi.newpipe.util;
+
+import java.util.Comparator;
+import java.util.List;
+
+public class ListUtils {
+    public static <T, P> int binarySearchUpperBound(List<T> ls, P target, Function<T, P> getter, Comparator<P> cmp) {
```

Comment:
Since this is a util method, you could add a Javadoc explaining inputs, return values, and optionally point out the complexity of the algorithm `O(log(n))`.